sources/api/ecs-settings-applier/src/main.rs

```diff
@@ -0,0 +1,158 @@
+#[macro_use]
+extern crate log;
+
+use std::{env, process};
+use std::fs;
+use std::path::{Path, PathBuf};
+use serde::{Serialize};
+use snafu::{ResultExt};
+use error::SettingsApplierError;
+
+const DEFAULT_API_SOCKET: &str = "/run/api.sock";
+const DEFAULT_ECS_CONFIG_PATH: &str = "/etc/ecs/ecs.config.json";
+const VARIANT_ATTRIBUTE_NAME: &str = "bottlerocket.variant";
+const VERSION_ATTRIBUTE_NAME: &str = "bottlerocket.version";
+
+
+#[derive(Serialize, Debug)]
+#[serde(rename_all="PascalCase")]
+struct ECSConfig {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    cluster: Option<String>,
+
+    #[serde(skip_serializing_if = "std::collections::HashMap::is_empty")]
+    instance_attributes: std::collections::HashMap<String, String>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    privileged_disabled: Option<bool>,
+}
+
+// Returning a Result from main makes it print a Debug representation of the error, but with Snafu
+// we have nice Display representations of the error, so we wrap "main" (run) and print any error.
+// https://github.com/shepmaster/snafu/issues/110
+fn main() {
+    if let Err(e) = run() {
+        eprintln!("{}", e);
+        process::exit(1);
+    }
+}
+
+fn run() -> Result<()> {
+    let args = parse_args(env::args());
+
+    // Get all settings values for config file templates
+    debug!("Requesting settings values");
+    let settings = schnauzer::get_settings(&args.socket_path).context(error::SettingsError)?;
+
+    debug!("settings = {:#?}", settings.settings);
+    let ecs = settings.settings.and_then(|s| s.ecs);
+    let ecs = if let Some(x) = ecs {
```

Comment:
I tried to see what other shapes this could take while adding some development changes:

https://github.com/jahkeup/bottlerocket/commit/15883b2eb3749d49d13446f36410629a39572068

There's a few other places in this code where the Option handling can be simplified because `None` isn't an error condition and is "Just data".

Reply:
I took your commit; thanks!